dht/table.go

```diff
@@ -93,12 +107,14 @@ func (table *InMemTable) AddPeer(peerID id.Signatory, peerAddr wire.Address) {
 
 	// Insert into the sorted address list based on its XOR distance from our
 	// own address.
-	i := sort.Search(len(table.sorted), func(i int) bool {
-		return table.isCloser(peerID, table.sorted[i])
-	})
-	table.sorted = append(table.sorted, id.Signatory{})
-	copy(table.sorted[i+1:], table.sorted[i:])
-	table.sorted[i] = peerID
+	if !ok {
```

Comment:
Why is this condition here? It prevents the address of a peer from ever being updated. I'm not sure that's the behaviour we want.

Reply:
This condition only stops re-adding a signatory to the sorted array of signatories, maintaining uniqueness of elements. The address can be replaced regardless and is done a couple lines before this conditional statement.